### PR TITLE
Fix source not being populated on component updates

### PR DESCRIPTION
**Pull request by chopkinsmade**
If you set the `source` property in the component like the below the dropdown would contain the values provided in the source property as expected:
```
uk_gov_dash_components.Dropdown(
        label="Comparison",
        id='comparison',
        source=["Something", "Something Else"],
        style={"minWidth": "50%"}
    )
```

image

However when using the dash `@app.callback` logic and changing the `source` property, the change in value was not being passed down to the underlying component. The reason for this is the props are being stored in state in the `Dropdown` but never refreshed after the initial load

### CODE REVIEW

File: usage.py [hunk 1]
```diff
@@ -25,6 +33,11 @@
 def display_output(value):
     return 'You have entered {}'.format(value)
 
+@app.callback(Output('comparison', 'source'), [Input('input', 'value')])
+def update_comparison(value):
+    print(value)
```
Comment (adrianclay) [inline usage.py:38]: Was this debugging/testing code?  Looks useful, but might be worth cleaning it up.

**Review by adrianclay — COMMENTED** [review]
I'm not very clear how this solves the issue I'm afraid.

Surely source was being provided previously anyway?

**Comment by chopkinsmade** [comment]
> I'm not very clear how this solves the issue I'm afraid.
> 
> Surely source was being provided previously anyway?
I've updated the PR description with a more detailed explanation

File: src/lib/fragments/Dropdown.react.js [hunk 2]
```diff
@@ -37,9 +37,11 @@ import {
  * @return {*}
  */
 const Dropdown = (props = {}) => {
-	const { label, ...autoComplete } = props;
+
+	const { label, source, ...autoComplete } = props;
 	const { labelProps, fieldProps } = useLabel(props);
 	const [autoCompleteProps] = useState(autoComplete);
```
Comment (chopkinsmade) [inline src/lib/fragments/Dropdown.react.js:43]: Its an issue with not resetting the state, fix this line

**Review by adrianclay — APPROVED** [review]
💥

**Pull request merged**